semantic_router/utils/function_call.py

```diff
@@ -6,38 +6,106 @@
 from semantic_router.llms import BaseLLM
 from semantic_router.schema import Message, RouteChoice
 from semantic_router.utils.logger import logger
+import re
 
 
-def get_schema(item: Union[BaseModel, Callable]) -> Dict[str, Any]:
-    if isinstance(item, BaseModel):
-        signature_parts = []
-        for field_name, field_model in item.__annotations__.items():
-            field_info = item.__fields__[field_name]
-            default_value = field_info.default
+def get_schemas(items: List[Union[BaseModel, Callable]]) -> List[Dict[str, Any]]:
+    schemas = []
+    for item in items:
```

Comment:
Can we keep `get_schema` function from before and call this `get_schema_list` and just have it contain the for loop and calls back to `get_schema`?

Reply:
Nice idea, yes. 

do you want me to do something similar of `get_schemas_openai`?

Reply:
whatever will produce the simplest, most intuitive, and cleanest code